pkg/cloudprovider/providers/azure/azure_loadbalancer.go

```diff
@@ -90,32 +95,34 @@ func (az *Cloud) GetLoadBalancer(clusterName string, service *v1.Service) (statu
 	}, true, nil
 }
 
-func (az *Cloud) determinePublicIPName(clusterName string, service *v1.Service) (string, error) {
+func (az *Cloud) determinePublicIPName(clusterName string, service *v1.Service) (pipName string, pipResourceGroup string, err error) {
 	loadBalancerIP := service.Spec.LoadBalancerIP
 	if len(loadBalancerIP) == 0 {
-		return getPublicIPName(clusterName, service), nil
+		return getPublicIPName(clusterName, service), "", nil
```

Comment:
Would it be safer to return `az.getPublicIPAddressResourceGroup()` here rather than `""`?  The current design puts the onus on the calling function to fill in the resource group in certain situations - this could lead to more code paths in the calling functions and more opportunities to miss something during testing.  (Longer term perhaps we need a better abstraction around this, such as a `PublicIPAddressReference`, but not sure about that yet!)

Reply:
I modified it so that I use `az.getPublicIPAddressResourceGroup()` as much as possible instead of passing the resource group around.